### PR TITLE
Adding PHPCR support

### PR DESCRIPTION
**Pull request by webda2l**
Follow https://github.com/FriendsOfSymfony/FOSElasticaBundle/pull/739

### CODE REVIEW

**Comment by lsmith77** [comment]
looks sensible from a quick inspection

/cc @dbu

File: Doctrine/PHPCR/ElasticaToDocumentTransformer.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                    ->getManager()
+                    ->findMany(null, $identifierValues)
+            );
+        }
+
+        // General case
+        return iterator_to_array(
+            $this->registry
+                ->getManager()
+                ->getRepository($this->objectClass)
```
Comment (dbu) [inline Doctrine/PHPCR/ElasticaToDocumentTransformer.php]: in terms of phpcr-odm, this means that objects of this class or extending classes will be found. which sounds like a good idea to me. (so just to make sure you are aware of that)

File: Doctrine/PHPCR/ElasticaToDocumentTransformer.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+     *
+     * @param array $identifierValues ids values
+     * @param Boolean $hydrate whether or not to hydrate the objects, false returns arrays
+     * @return array of objects or arrays
+     */
+    protected function findByIdentifiers(array $identifierValues, $hydrate)
+    {
+        // Special case where model is interface. But maybe remove because negative performance impact
+        $reflectionClass = new \ReflectionClass($this->objectClass);
+        if ($reflectionClass->isInterface()) {
```
Comment (lsmith77) [inline Doctrine/PHPCR/ElasticaToDocumentTransformer.php]: maybe http://php.net/manual/en/function.interface-exists.php is more efficient .. or get_declared_interfaces() .. though the later I doubt is more efficient.

**Comment by merk** [comment]
Not knowing much about PHPCR, I'll defer to @dbu and @lsmith77's evaluations.

I am nervous about merging as is: without any integration tests it isnt possible for me to know if any changes will break this code. I realise there are few tests for other backends but we must start somewhere.

**Comment by dbu** [comment]
it looks fine, but a few tests would sure make sense. @merk can you point @webda2l to some examples how to do this?

a short note in the doc would also make sense so people actually detect this exists.

File: Doctrine/PHPCR/ElasticaToDocumentTransformer.php (hunk 1 above)
Comment (webda2l) [inline Doctrine/PHPCR/ElasticaToDocumentTransformer.php]: Yes I'm aware, thks

File: Doctrine/PHPCR/ElasticaToDocumentTransformer.php (hunk 2 above)
Comment (webda2l) [inline Doctrine/PHPCR/ElasticaToDocumentTransformer.php]: Maybe I'm wrong, but theses methods doesn't seems fit the needs... oO

If I want avoid this Reflection use, but allow such ElasticableInterface use for multiple models in foselastica settings, I could do differently however.
Check in a higher level, during foselastica init configuration, if the model value is an interface case and pass this information to the ElasticaToModelTransformer by example.
But see if its has plus-value.
Or I can also remove this special case and do it only for my project.

File: Doctrine/PHPCR/ElasticaToModelTransformer.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+     *
+     * @param array $identifierValues ids values
+     * @param Boolean $hydrate whether or not to hydrate the objects, false returns arrays
+     * @return array of objects or arrays
+     */
+    protected function findByIdentifiers(array $identifierValues, $hydrate)
+    {
+        // Special case where model is interface
+        $reflectionClass = new \ReflectionClass($this->objectClass);
+        if ($reflectionClass->isInterface()) {
```
Comment (dbu) [inline Doctrine/PHPCR/ElasticaToModelTransformer.php]: i think [interface_exists](http://php.net/manual/en/function.interface-exists.php) is exactly what you want here.

**Comment by merk** [comment]
As a starting point: duplicating some of the code under Tests/Functional that relates to the ORM kernel configuration would be great.

**Comment by webda2l** [comment]
This is limited in written tests.. but this PR could be RFR.

**Comment by dbu** [comment]
this looks good to me. what about merging it and then trying to bootstrap some more thorough functional testing for the whole bundle?

**Comment by Buri** [comment]
Any progress here?

**Comment by dbu** [comment]
oh, there was no comment after the updated commits so i got no notification. @merk this looks good to me (but i did not run the code, just glance over it). imho we can merge this.

**Comment by dymissy** [comment]
Any news about the PR?

**Comment by dbu** [comment]
@merk are you still active on this bundle?

**Comment by dymissy** [comment]
If could help someone, I tried the PR code on my environment and it works fine. Contents are indexed correctly. I just have to find a solution for multi-language documents but if the PR will be approved I could try to add a new one in the next month.

**Comment by merk** [comment]
I am not particularly active at the moment because my current projects dont have searching capabilities and I cant spare the time.

I'm happy for other FOS team members to help out if they have time. I'd love to do some major refactoring since I'm not happy with the way the bundle is used but it'll have to wait.

**Comment by dbu** [comment]
okay, i see. its the same for me. i was only on this branch because of PHPCR. i guess we can merge this PR, ok?

**Comment by merk** [comment]
My only concern is taking on more code becomes a liability if noone will maintain it (i dont use PHPCR). Other than that, happy to merge

**Comment by dbu** [comment]
i can help with questions/problems about the PHPCR-ODM integration.

**Comment by dbu** [comment]
@webda2l thanks a lot!
i notice we miss documentation. @dymissy or @webda2l could one of you do a pull request to add this to https://github.com/FriendsOfSymfony/FOSElasticaBundle/blob/master/Resources/doc/types.md#realtime-selective-index-update ? (and maybe check the doc for other mentions of doctrine orm and add phpcr odm as another option there?)

**Comment by dymissy** [comment]
@dbu I just created a new PR here https://github.com/FriendsOfSymfony/FOSElasticaBundle/pull/966

While I'm using the library I will try to update the doc with some example using PHPCR.

**Pull request merged**